src/main.rs

```diff
@@ -15,23 +16,28 @@ const SLEEP_TIME_MS: u64 = 50;
 const MINIDSP_GAIN_MIN: f64 = -127.0;
 const MINIDSP_GAIN_MAX: f64 = 0.0;
 
+// Lazy static needed since Regex::new() is a function call
+lazy_static! {
+    static ref GAIN_REGEX: Regex =
+        Regex::new(r"Gain\((?P<gain>-*\d*\.*\d*)\)").expect("Failed to create regex for gain");
+}
+
 fn get_gain() -> f64 {
     let output = Command::new(MINIDSP_BINARY)
         .output()
         .expect("Failed to retrieve current gain");
     let output_string =
         String::from_utf8(output.stdout).expect("Failed to convert output to string");
 
-    // Fetch gain through regex
-    let gain_regex =
-        Regex::new(r"Gain\((?P<gain>-*\d*\.*\d*)\)").expect("Failed to create regex for gain");
-    let capture_groups = gain_regex
+    // Capture gain through regex
+    let capture_groups = GAIN_REGEX
         .captures(&output_string)
         .expect("Failed to retrieve gain from regex");
-    let gain = &capture_groups["gain"]
+
+    // Return the gain as a floating number
+    capture_groups["gain"]
         .parse::<f64>()
-        .expect("Failed to convert gain to float");
-    *gain
+        .expect("Failed to convert gain to float")
 }
 
 fn apply_gain(gain: &str) {
```

Comment:
I think println! is fine here. stdout will be redirected to syslog when using systemd. Not really targeting Windows here.

Reply:
Sick moves. This is why low-level confuses me ❤️
Logger is std in jvm